narps_open/pipelines/matlabbatch_UK24.m

```diff
@@ -0,0 +1,292 @@
+
+% first-level job ----
+
+% $ narps_description -t UK24 -d preprocessing --json
+
+
+clear matlabbatch;
+
+% We consider that anat and func data have been gunzipped first
+
+% % gunzip anat
+% matlabbatch{1}.cfg_basicio.file_dir.file_ops.cfg_gunzip_files.files = {'/Users/camaumet/Softs/narps_open_pipelines/data/original/ds001734/sub-001/anat/sub-001_T1w.nii.gz'};
+% matlabbatch{1}.cfg_basicio.file_dir.file_ops.cfg_gunzip_files.outdir = {''};
+% matlabbatch{1}.cfg_basicio.file_dir.file_ops.cfg_gunzip_files.keep = true;
+% 
+% % gunzip sbref
+% matlabbatch{end+1}.cfg_basicio.file_dir.file_ops.cfg_gunzip_files.files = {'/Users/camaumet/Softs/narps_open_pipelines/data/original/ds001734/sub-001/anat/sub-001_T1w.nii.gz'};
+% matlabbatch{end}.cfg_basicio.file_dir.file_ops.cfg_gunzip_files.outdir = {''};
+% matlabbatch{end}.cfg_basicio.file_dir.file_ops.cfg_gunzip_files.keep = true;
+
+% 1 - Coregistration of the structural T1w image to the reference 
+% functional image (defined as the single volume 'sbref' image acquired 
+% before the first functional run) \n
+matlabbatch{1}.spm.spatial.coreg.estimate.ref(1) = {...
+    '/ABS_PATH/data/original/ds001734/sub-001/func/sub-001_task-MGT_run-01_sbref.nii,1'}; %SBREF_FILE
+matlabbatch{end}.spm.spatial.coreg.estimate.source(1) = {...
+    '/ABS_PATH/data/original/ds001734/sub-001/anat/sub-001_T1w.nii,1'}; %ANAT_FILE
+
+% 2 - Segmentation of the coregistered T1w image into grey matter, white 
+% matter and cerebrospinal fluid tissue maps.\n
+
+% note le fichier ANAT va garder le meme nom mais en pratique il y a une dependance sur la sortie du coreg
+matlabbatch{end+1}.spm.spatial.preproc.channel.vols = {...
+    '/ABS_PATH/data/original/ds001734/sub-001/anat/sub-001_T1w.nii,1'}; % this is the coregistered ANAT
+
+% --> output ANAT_SEG_PARAM
+
+% 3 - Reslicing of coregistered T1w image and segmentations to the same 
+% voxel space of the reference (sbref) image
+
+% Note Camille: I am not sure how to do this with the matlabbatch (imcalc
+% should work but maybe there is a more direct option)
+
+% spm_reslice{{ANAT, c1ANAT, c2ANAT, c3ANAT}, SBREF}
+
+% 4 - Functional MRI volume realignment: first realigning all 4 'sbref' 
+% images to the one acquired before the first run; followed by realignment 
+% of all volumes in a particular run to the first image in that run. 
```

Comment:
Here, `first image in that run` does not refer to the sbref file, right ?

Reply:
Agreed. Though the first image in the run and the sbref should be very close.